diff --git a/node_modules/@isomorphic-git/cors-proxy/index.js b/node_modules/@isomorphic-git/cors-proxy/index.js
index 68241c8..a29ffad 100644
--- a/node_modules/@isomorphic-git/cors-proxy/index.js
+++ b/node_modules/@isomorphic-git/cors-proxy/index.js
@@ -16,9 +16,7 @@ async function service (req, res) {
       <html>
         <title>@isomorphic-git/cors-proxy</title>
         <h1>@isomorphic-git/cors-proxy</h1>
-        <p>This is the server software that runs on <a href="https://cors.isomorphic-git.org">https://cors.isomorphic-git.org</a>
-           &ndash; a free service (generously sponsored by <a href="https://www.clever-cloud.com/?utm_source=ref&utm_medium=link&utm_campaign=isomorphic-git">Clever Cloud</a>)
-           for users of <a href="https://isomorphic-git.org">isomorphic-git</a> that enables cloning and pushing repos in the browser.</p>
+        <p>This is the server software for users of <a href="https://isomorphic-git.org">isomorphic-git</a> that enables cloning and pushing repos in the browser.</p>
         <p>The source code is hosted on Github at <a href="https://github.com/isomorphic-git/cors-proxy">https://github.com/isomorphic-git/cors-proxy</a></p>
         <p>It can also be installed from npm with <code>npm install <a href="https://npmjs.org/package/${pkg.name}">@isomorphic-git/cors-proxy</a></code></p>
 
