diff --git a/builders/docker_builder.py b/builders/docker_builder.py
index f6a4177..5f504cf 100644
--- a/builders/docker_builder.py
+++ b/builders/docker_builder.py
@@ -36,7 +36,7 @@ except ModuleNotFoundError:
     pass

 ANSI_ESCAPE = re.compile(r"\x1B\[[0-?]*[ -/]*[@-~]")
-DOCKER_API_VERSION = "1.35"
+DOCKER_MIN_API_VERSION = os.getenv("DOCKER_MIN_API_VERSION", "1.44")


 class DockerBuilder(OCIBuilder):
@@ -187,7 +187,7 @@ class DockerBuilder(OCIBuilder):
         # image for squashing.
         timeout = parse_env_timeout("DOCKER_TIMEOUT", "600")

-        params = {"version": DOCKER_API_VERSION}
+        params = {"version": DOCKER_MIN_API_VERSION}
         params.update(docker.utils.kwargs_from_env())
         params["timeout"] = timeout
